File: lib/java/main/src/main/java/alphaTab/alphaSkia/AlphaSkiaNative.java
package alphaTab.alphaSkia;

/**
 * The base class for AlphaSkia objects wrapping native Skia objects.
 */
public abstract class AlphaSkiaNative implements AutoCloseable {
    static {
        if(!AlphaSkiaPlatform.isNativeLibLoaded()) {
            throw new IllegalStateException("Initialize the alphaSkia platform first");
        }
    }

    long handle;

    AlphaSkiaNative(long handle) {
        this.handle = handle;
    }


    @Override
    public abstract void close() throws Exception;
}

File: lib/kotlin/mpp/src/commonMain/kotlin/alphaTab/alphaSkia/AlphaSkiaNative.kt
package alphaTab.alphaSkia;

/**
 * The base class for AlphaSkia objects wrapping native Skia objects.
 */
abstract class AlphaSkiaNative internal constructor(internal var handle: Long) : AutoCloseable {
    companion object {
        init {
            if (!AlphaSkiaPlatform.isNativeLibLoaded()) {
                throw IllegalStateException("Initialize the alphaSkia platform first")
            }
        }
    }

    abstract override fun close()
}
